### PR TITLE
Support to print lists in REPL

### PR DESCRIPTION
**Pull request by Vipul-Cariappa**

### CODE REVIEW

**Comment by Vipul-Cariappa** [comment]
cc @Shaikh-Ubaid, could you please have a look at it? Does it require any more changes?

File: src/libasr/pass/global_stmts.cpp [hunk 1]
```diff
@@ -141,7 +82,47 @@ void pass_wrap_global_stmts(Allocator &al,
...
     if (return_var) {
         // The last defined `return_var` is the actual return value
+        LCOMPILERS_ASSERT(type)
+        LCOMPILERS_ASSERT(return_var_ref)
+
         ASR::down_cast2<ASR::Variable_t>(return_var)->m_intent = ASRUtils::intent_return_var;
+        std::string global_underscore_name = "_" + fn_name_s;
+        s.from_str(al, global_underscore_name);
+        
+        ASR::asr_t *global_underscore = ASR::make_Variable_t(al, loc,
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/global_stmts.cpp:92]: Are we supporting `lists` in this PR or the `_` variable? If we are adding support for both, can you please break this into two PR with adding tests for each in the respective PRs?

File: src/lpython/python_evaluator.cpp [hunk 2]
```diff
@@ -319,4 +360,105 @@ Result<std::unique_ptr<LLVMModule>> PythonCompiler::get_llvm3(
 #endif
 }
 
+void print_type(ASR::ttype_t *t, void *data, std::string &result);
+
+std::string PythonCompiler::string_aggregate_type(const struct EvalResult &r) {
```
Comment (Shaikh-Ubaid) [inline src/lpython/python_evaluator.cpp:365]: Is this function converting a given aggregate type to string? If yes, it should be called `aggregate_type_to_string()` or `cnvrt_aggregate_type_to_string()` or something similar. `string_aggregate_type` doesn't give any good idea of what the function does and seems rather misleading.

File: src/libasr/pass/global_stmts.cpp [hunk 3]
```diff
@@ -141,7 +82,47 @@ void pass_wrap_global_stmts(Allocator &al,
...
+        //             ASR::storage_typeType::Default, type_pointer,
+        //             nullptr, ASR::abiType::BindC,
+        //             ASR::Public, ASR::presenceType::Required, false);
+        //     return_var_ref = EXPR(ASR::make_Var_t(al, loc, down_cast<ASR::symbol_t>(return_var)));
+        //     ASR::stmt_t* asr_stmt = ASRUtils::STMT(ASR::make_Assignment_t(al, loc, return_var_ref, value, nullptr));
+        //     body.push_back(al, asr_stmt);
+        //     idx++;
+        //     return_var_ref = nullptr;
+        //     return_var = nullptr;
+        // }
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/global_stmts.cpp:125]: Commented code should be removed.

File: src/lpython/python_evaluator.cpp [hunk 4]
```diff
@@ -119,10 +122,40 @@ Result<PythonCompiler::EvalResult> PythonCompiler::evaluate(
...
-        call_run_fn = true;
-        return_type = m->get_return_type(run_fn);
+    std::string return_type = m->get_return_type(run_fn);
+    if (return_type != "none") {
+      call_run_fn = true;
+    }
+
+    llvm::Type *type = nullptr;
+    ASR::symbol_t *f = symbol_table->get_symbol("_" + run_fn);
+    if ((return_type == "struct") && (f)) {
```
Comment (Shaikh-Ubaid) [inline src/lpython/python_evaluator.cpp:132]: Slightly confused here. What are we processing inside this `if` and why do we need this processing?

**Review by Shaikh-Ubaid — COMMENTED** [review]
Shared few comments. Rest seems good.

File: src/lpython/python_evaluator.cpp [hunk 5]
```diff
@@ -319,4 +344,136 @@ Result<std::unique_ptr<LLVMModule>> PythonCompiler::get_llvm3(
...
+    size_t *offsets = r.structure.offsets;
+    size_t element_size = r.structure.element_size;
+
+    std::string result;
+
+    if (asr_type->type == ASR::ttypeType::List) {
+        int32_t size = *(int32_t*)(((char*)data)+offsets[0]);
+        void *array = *(void**)(((char*)data)+offsets[2]);
+        // int32_t capacity = *(int32_t*)(((char*)data)+offsets[1]);
+        // std::cout << "size: " << size << " capacity: " << capacity << " array: " << array << std::endl;
```
Comment (Shaikh-Ubaid) [inline src/lpython/python_evaluator.cpp:361]:
```suggestion
```

File: src/lpython/python_evaluator.cpp [hunk 6]
```diff
@@ -319,4 +344,134 @@ Result<std::unique_ptr<LLVMModule>> PythonCompiler::get_llvm3(
 #endif
 }
 
+void print_type(ASR::ttype_t *t, void *data, std::string &result);
```
Comment (Shaikh-Ubaid) [inline src/lpython/python_evaluator.cpp:347]: Do we need this declaration? If we need it then a better place for it could be `python_evaluator.h`.

**Review by Shaikh-Ubaid — APPROVED** [review]
Looks good to me. Great work @Vipul-Cariappa! Left a minor comment. It can be fixed in a separate PR.

**Comment by Shaikh-Ubaid** [comment]
@Vipul-Cariappa please resolve the conflicts and we can merge this.

**Pull request merged**